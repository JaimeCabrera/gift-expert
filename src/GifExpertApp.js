import React, { useState } from 'react'
import AddCategory from './components/AddCategory';
import GifGrid from './components/GifGrid';

const GifExpertApp = () => {

   const [categories, setCategories] = useState(['simpson']);

   // const handleAdd = () => {
   //    // para aÃ±adir nuevos elementos al array usamos el op spreed [...ArrayAnterior,nuevo elemento]
   //    // setCategories([...categories, 'futurama']);
   //    // otra forma utilizando la funcion callback
   //    setCategories(cats =>[...cats,'Futurama']);

   // }
   return (
      <>
         <h2>Gif Expert App</h2>
         <AddCategory setCategories={setCategories}/>
         <hr />
         {/* <button onClick={handleAdd}>Agregar</button> */}
         <ol>
            {
               categories.map(category => (
                  <GifGrid
                     key={category}
                     category={category}
                  />
               ))
            }
         </ol>
      </>
   )
}

export default GifExpertApp
