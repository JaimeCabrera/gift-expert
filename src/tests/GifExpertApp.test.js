import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { GifExpertApp } from '../GifExpertApp';

describe('Pruebas en el componente <GifExpertApp />', () => {
   

   test('Debe renderizar correctamente el componente', () => {
      const wrapper = shallow(<GifExpertApp />);
      expect(wrapper).toMatchSnapshot();
   });

   test('debe demostrar una lista de categorias o <GifGrids/>', () => {
      const categories = ['one punch', 'dragon bz'];
      const wrapper = shallow(<GifExpertApp defaultCategory={categories}/>);
      // expect(wrapper.find('GifGrid').length).toBe(categories.length);
   })
   
   

})
