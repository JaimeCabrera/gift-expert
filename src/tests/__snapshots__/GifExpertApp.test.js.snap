// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <GifExpertApp /> Debe renderizar correctamente el componente 1`] = `
<Fragment>
  <h2>
    Gif Expert App
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
